test/Actions.test.ts

```diff
@@ -0,0 +1,121 @@
+import {MockControllerInstance} from '../build/types/truffle-types'
+
+const {BN, expectRevert} = require('@openzeppelin/test-helpers')
+
+const MockController = artifacts.require('MockController.sol')
+
+contract('Actions', ([owner, random]) => {
+  // Controller mock instance
+  let controller: MockControllerInstance
+  const ZERO_ADDR = '0x0000000000000000000000000000000000000000'
+
+  before('Deployment', async () => {
+    controller = await MockController.new({from: owner})
+  })
+
+  describe('Parse Deposit Arguments', () => {
+    it('should not be able to parse a non Deposit action', async () => {
+      const actionType = '0'
+      const asset = ZERO_ADDR
+      const vaultId = '0'
+      const amount = '10'
+      const index = '0'
+      const bytesArgs = ZERO_ADDR
+
+      const data = {
+        actionType: actionType,
+        owner: owner,
+        sender: owner,
+        asset: asset,
+        vaultId: vaultId,
+        amount: amount,
+        index: index,
+        data: bytesArgs,
+      }
+
+      await expectRevert(
+        controller.testParseDespositAction(data),
+        'Actions: can only parse arguments for deposit actions',
+      )
+    })
+    it('should not be able to parse an invalid sender address', async () => {
+      const actionType = '3'
```

Comment:
Given that we will have so many actions in this file in the future, will be better to define a Enum now.

Reply:
hmm what do you mean?

Reply:
Instead of using 
```javascript
const actionType = '3'
```
Have 
```javascript
// Defined globally
enum ActionType {
  OpenVault,
  MintShort,
  BurnShort,
  Deposit 
}

// In each test
const actionType = ActionType.Deposit
```

Reply:
ohh you mean in the test file?